src/core.js

```diff
@@ -227,6 +227,8 @@ const alignWithDOM = function(nodeName, key) {
     const keyNode = keyMap[key];
     if (matches(nodeName, key, keyNode)) {
       node = keyNode;
+    } else if (keyNode && keyNode !== currentNode) {
+      removeChild(currentParent, keyNode, keyMap);
```

Comment:
I see, I thought this was already done below by the `replaceChild` call.

Since the removal is being handled there, I think it'd probably be best to add the `markDeleted` call there too instead of here. Do you agree?